API.md

```diff
@@ -652,7 +652,7 @@ For example, use `is: Joi.number().required()` to guarantee that a joi reference
 
 ```js
 const schema = {
-    a: Joi.any().valid('x').when('b', { is: 5, then: Joi.valid('y'), otherwise: Joi.valid('z') }),
```

Comment:
Why the change here? I like adding `exist` to the docs, but I'm not sure why the specific value `5` was essentially changed to `Joi.required()`.

I know it's just an example in the documentation, but still. :P